packages/devextreme/js/__internal/ui/list/m_list.edit.decorator.switchable.ts

```diff
@@ -136,80 +140,91 @@ class SwitchableEditDecorator extends EditDecorator {
+  _cacheReadyToDeleteItem($itemElement: dxElementWrapper, cache: boolean): void {
     if (cache) {
       this._$readyToDeleteItem = $itemElement;
     } else {
       delete this._$readyToDeleteItem;
     }
   }
 
-  _animateToggleDelete($itemElement, readyToDelete): void {
+  _animateToggleDelete($itemElement: dxElementWrapper, readyToDelete: boolean): void {
     if (readyToDelete) {
       this._enablePositioning($itemElement);
       this._prepareDeleteReady($itemElement);
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
       this._animatePrepareDeleteReady($itemElement);
       eventsEngine.off($itemElement, pointerEvents.up);
     } else {
       this._forgetDeleteReady($itemElement);
-      // @ts-expect-error ts-error
-      this._animateForgetDeleteReady($itemElement).done(this._disablePositioning.bind(this, $itemElement));
+      this._animateForgetDeleteReady($itemElement)
+        // @ts-expect-error ts-error
+        .done(this._disablePositioning.bind(this, $itemElement));
     }
   }
 
-  _enablePositioning($itemElement): void {
+  _enablePositioning($itemElement: dxElementWrapper): void {
     $itemElement.addClass(SWITCHABLE_MENU_ITEM_SHIELD_POSITIONING_CLASS);
     eventsEngine.on($itemElement, ACTIVE_EVENT_NAME, noop);
-    eventsEngine.one($itemElement, pointerEvents.up, this._disablePositioning.bind(this, $itemElement));
+    eventsEngine.one(
+      $itemElement,
+      pointerEvents.up,
+      this._disablePositioning.bind(this, $itemElement),
+    );
   }
 
-  _disablePositioning($itemElement): void {
+  // eslint-disable-next-line class-methods-use-this
+  _disablePositioning($itemElement: dxElementWrapper): void {
     $itemElement.removeClass(SWITCHABLE_MENU_ITEM_SHIELD_POSITIONING_CLASS);
     eventsEngine.off($itemElement, ACTIVE_EVENT_NAME);
   }
 
-  _prepareDeleteReady($itemElement): void {
+  // eslint-disable-next-line class-methods-use-this
+  _prepareDeleteReady($itemElement: dxElementWrapper): void {
     $itemElement.addClass(SWITCHABLE_DELETE_READY_CLASS);
   }
 
-  _forgetDeleteReady($itemElement): void {
+  // eslint-disable-next-line class-methods-use-this
+  _forgetDeleteReady($itemElement: dxElementWrapper): void {
     $itemElement.removeClass(SWITCHABLE_DELETE_READY_CLASS);
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  _animatePrepareDeleteReady($element: dxElementWrapper): void {
-    Class.abstract();
-  }
+  abstract _animatePrepareDeleteReady($element: dxElementWrapper): Promise<void>;
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  _animateForgetDeleteReady($itemElement?): void {
-    Class.abstract();
-  }
+  abstract _animateForgetDeleteReady($itemElement?: dxElementWrapper): Promise<void>;
 
-  _getDeleteButtonContainer($itemElement) {
-    $itemElement = $itemElement || this._$readyToDeleteItem;
+  _getDeleteButtonContainer($itemElement: dxElementWrapper): dxElementWrapper {
```

Comment:
According to the code, argument is optional
```suggestion
  _getDeleteButtonContainer($itemElement?: dxElementWrapper): dxElementWrapper {
```